packages/axelar-gmp-contract/docs/remote-account-architecture.md

```diff
@@ -0,0 +1,186 @@
+
+The `EvmAccountKit` is a core orchestrated object that encapsulates the state and behavior for managing communication with a remote EVM chain via the Axelar GMP protocol. It handles sending and receiving messages, tracking remote account creation, and providing invitations for user actions.
+
+Some key components:
+
+- **tap.receiveUpcall(event)**
+
+  Processes incoming IBC transfer events. It does the following:
+
+  - Reads and interprets the message content
+  - Checks the result of any contract calls to EVM that were performed
+  - Saves the remote Ethereum account address when it becomes available
+
+- **holder.sendGmp(seat, offerArgs)**
+
+  Responsible for performing cross-chain token transfers and invoking contract calls on the remote EVM chain via Axelar GMP.
+
+- **holder.getLocalAddress()**
+
+  Returns the Bech32 address of the LCA.
+
+- **holder.getAddress()**
+
+  Returns the remote EVM account address (if created).
+
+- **holder.getLatestMessage()**
+
+  Returns the latest contract call result received from the EVM chain.
+
+---
+
+## 2. Ethereum Chain
+
+Two smart contracts are deployed on Ethereum to enable remote account operations from Agoric: `Factory` and `Wallet`.
+
+### Factory Contract
+
+**Defined in:** [`Factory.sol`](../solidity/contracts/Factory.sol)
+
+The `Factory` contract is responsible for creating new instances of the `Wallet` contract. It inherits `AxelarExecutable` to receive interchain requests and reply back to Agoric using Axelar's General Message Passing (GMP). It stores the gateway and gas receiver addresses, and keeps the local chain name for reference. It uses Axelar's [`IAxelarGasService`](https://docs.axelar.dev/dev/gas-service/intro/#introduction-to-axelar-gas-service-contract) to pay for outbound message gas, and emits events when wallets are created.
+
+#### Key Functions
+
+- **createVendor(string memory owner)**
+
+  - Deploys a new `Wallet` contract.
+  - Sets the provided `owner` as the wallet's owner.
+  - Emits the `WalletCreated` event with the new wallet address and `owner` string.
+
+- **\_execute**
+
+  - Triggered by Axelar when a remote call from Agoric is received.
+  - Uses `sourceAddress` (the Agoric caller) as the owner to call `createVendor`.
+  - Encodes the resulting `Wallet` address into an `AgoricResponse` struct.
+  - Sends the result back to the Agoric chain using `_send`.
+
+- **\_send**
+
+  - Calls `payNativeGasForContractCall` to ensure Axelar relayers process the outbound message.
+  - Uses `callContract` to send the response back to Agoric.
+
+---
+
+### Wallet Contract
+
+**Defined in:** [`Factory.sol`](../solidity/contracts/Factory.sol)
+
+The `Wallet` contract acts as a remote Ethereum account controlled by Agoric. It receives messages from Agoric and performs on-chain contract calls. It inherits from `AxelarExecutable` to receive interchain messages. The contract also integrates with `IAxelarGasService` for paying gas fees when sending responses back to Agoric.
```

Comment:
"It receives messages from Agoric" hm... that's what it's designed to do, but I wonder whether this phrasing makes us forget that anybody can send messages to it (though those messages will fail if `msg.sender` isn't right).